test/RequestStreamTest.cpp

```diff
@@ -120,3 +145,29 @@ TEST(RequestStreamTest, RequestOnDisconnectedClient) {
   CHECK_WAIT(wait_for_on_error);
   ASSERT(did_call_on_error);
 }
+
+class TestHandlerError : public rsocket::RSocketResponder {
+ public:
+  Reference<Flowable<Payload>> handleRequestStream(
+      Payload request,
+      StreamId) override {
+    // string from payload data
+    auto requestString = request.moveDataToString();
+
+    return Flowables::error<Payload>(folly::exception_wrapper(
+        std::runtime_error("A wild error appeared")));
+  }
+};
+
+TEST(RequestStreamTest, HandleError) {
+  folly::ScopedEventBaseThread worker;
+  auto server = makeServer(std::make_shared<TestHandlerError>());
+  auto client = makeClient(worker.getEventBase(), *server->listeningPort());
+  auto requester = client->getRequester();
+  auto ts = TestSubscriber<std::string>::create();
+  requester->requestStream(Payload("Bob"))
+           ->map([](auto p) { return p.moveDataToString(); })
```

Comment:
I wish our TestSubscriber consumed Payloads rather than strings :/ we end up testing MapOperator all the time in addition to the request/response path. Could be worthwhile to do in a later change.